client/session.go

```diff
@@ -1306,6 +1323,11 @@ func (s *session) FetchBlocksFromPeers(
 	metadatas []block.ReplicaMetadata,
 	opts result.Options,
 ) (PeerBlocksIter, error) {
+	nsMetadata, err := s.namespaceMetadata(namespace)
```

Comment:
I think instead of polluting the client with the namespace registry it would be best to simply make this method have the following method signature:
```
FetchBlocksFromPeers(
	namespace namespace.Metadata,
	shard uint32,
	metadatas []block.ReplicaMetadata,
	opts result.Options,
) (PeerBlocksIter, error)
```